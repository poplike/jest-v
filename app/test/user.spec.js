/**
 * 测试内容包括以下
 * 挂载子组件-stubs
 * 测试el-button
 */

import { mount, createLocalVue } from '@vue/test-utils';
import User from '@/views/user.vue';
import ElementUI from 'element-ui';

// 创建临时Vue实例，挂载组件中使用的插件
const localVue = createLocalVue();
localVue.use(ElementUI);

describe('user', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = mount(User, {
      localVue
    });
  });

  afterEach(() => {
    wrapper.destroy();
  });

  it('message', () => {
    expect(wrapper.vm.message).toBe('name');
  });

  it('handleClick', () => {
    let btn = wrapper.find('.el-button');
    btn.trigger('click');
    expect(wrapper.vm.message).toBe('new name');
  });

  it('onBtnClick', () => {
    wrapper.vm.onBtnClick()
    expect(wrapper.vm.message).toBe('new name bn1')
  });

  it('显示时snapshot测试', () => {
    // 方便查找dom
    expect(wrapper.vm.$el).toMatchSnapshot();
  });
});
