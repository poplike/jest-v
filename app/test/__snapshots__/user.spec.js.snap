// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`user 显示时snapshot测试 1`] = `
<div>
  
  name
  
  <div>
    123
  </div>
   
  <button
    class="el-button el-button--default"
    type="button"
  >
    <!---->
    <!---->
    <!---->
  </button>
</div>
`;
